src/hotspot/share/opto/compile.cpp

```diff
@@ -1118,7 +1121,7 @@ void Compile::verify_start(StartNode* s) const {
  * the ideal graph.
  */
 StartNode* Compile::start() const {
-  assert (!failing(), "Must not have pending failure. Reason is: %s", failure_reason());
+  assert (!failing_internal() || C->failure_is_artificial(), "Must not have pending failure. Reason is: %s", failure_reason());
```

Comment:
Having the stress mode in debug builds requires weakening asserts since debug builds assert these paths are not taken.